engine/env/spark/src/main/java/org/datacleaner/spark/Main.java

```diff
@@ -39,20 +56,61 @@ public static void main(String[] args) {
         final String confXmlPath = args[0];
         final String analysisJobXmlPath = args[1];
 
-        final String propertiesPath; 
+        final String propertiesPath;
         if (args.length > 2) {
             propertiesPath = args[2];
         } else {
             propertiesPath = null;
         }
 
-        final SparkJobContext sparkJobContext = new SparkJobContext(sparkContext, confXmlPath, analysisJobXmlPath, propertiesPath);
-
+        final SparkJobContext sparkJobContext = new SparkJobContext(sparkContext, confXmlPath, analysisJobXmlPath,
+                propertiesPath);
         final SparkAnalysisRunner sparkAnalysisRunner = new SparkAnalysisRunner(sparkContext, sparkJobContext);
         try {
-            sparkAnalysisRunner.run();
+            final AnalysisResultFuture result = sparkAnalysisRunner.run();
+            if (result.isDone()) {
+                final Map<ComponentJob, AnalyzerResult> resultMap = result.getResultMap();
+                final SimpleAnalysisResult simpleAnalysisResult = new SimpleAnalysisResult(resultMap,
+                        result.getCreationDate());
+                final String resultJobFilePath = getResultJobFilePath(sparkContext, sparkJobContext);
+                logger.info("The result of the job was written to " + resultJobFilePath);
+                if (resultJobFilePath != null) {
+                    final HdfsResource hdfsResource = new HdfsResource(resultJobFilePath);
+                    final OutputStream out = hdfsResource.write();
+                    try {
+                        SerializationUtils.serialize(simpleAnalysisResult, out);
+                    } catch (SerializationException e) {
+                        logger.error("Error while trying to serialize the job");
+                        throw e;
+                    } finally {
+                        FileHelper.safeClose(out);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Exception " + e.getStackTrace());
+            throw e;
         } finally {
             sparkContext.stop();
         }
     }
+
+    private static String getResultJobFilePath(final JavaSparkContext sparkContext,
+            final SparkJobContext sparkJobContext) {
+        String resultPath = sparkJobContext.getResultPath();
+        final Configuration hadoopConfiguration = sparkContext.hadoopConfiguration();
+        final String fileSystemPrefix = hadoopConfiguration.get("fs.defaultFS");
+        if (resultPath == null) {
+            resultPath = fileSystemPrefix + DEFAULT_RESULT_PATH;
+        } else {
+            resultPath = fileSystemPrefix + resultPath;
```

Comment:
Hmmm... Wont this break with URIs containg the host now (these may be needed by customers using [federated namenodes](https://hadoop.apache.org/docs/r2.4.1/hadoop-project-dist/hadoop-hdfs/Federation.html))? 

(It is pretty simple to test if it does, using the URI class' isAbsolute() method)